### PR TITLE
Fix messaging examples

### PR DESCRIPTION
**Pull request by gewenyu99**
## What does this PR do?

Fixes: https://github.com/appwrite/website/issues/744

## Test Plan

Manual

## Related PRs and Issues

* https://github.com/appwrite/website/issues/744

### Have you read the [Contributing Guidelines on issues](https://github.com/appwrite/appwrite/blob/master/CONTRIBUTING.md)?

Yes

### CODE REVIEW

File: src/routes/docs/products/messaging/topics/+page.markdoc [hunk 1]
```diff
@@ -303,54 +292,34 @@ If you can't find the targets you'd like to add, see the [targets page](/docs/pr
...
-        '[SUBSCRIBER_ID]',                                 // subscriberId
-        '[TARGET_ID]'                                      // targetId
-    );
-
-promise.then(function (response) {
-    console.log(response);
-}, function (error) {
-    console.log(error);
-});
+const subscriber = await messaging.createSubscriber(
```
Comment (stnguyen90) [inline src/routes/docs/products/messaging/topics/+page.markdoc:302]: Seems like there's some inconsistency with when we use `await` as it's not used in the examples above 🧐

File: src/routes/docs/products/messaging/topics/+page.markdoc [hunk 2]
```diff
@@ -434,72 +397,53 @@ Subscriber result = await messaging.CreateSubscriber(
...
 
-  client
-    .setEndpoint('https://cloud.appwrite.io/v1') // Your API Endpoint
-    .setProject('5df5acd0d48c2')                           // Your project ID
-    .setJWT('eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJ...') // Your secret JSON Web Token
-  ;
+Messaging messaging = Messaging(client);
 
-  Future result = messaging.createSubscriber(
+Subscriber subscriber result = await messaging.createSubscriber(
```
Comment (stnguyen90) [inline src/routes/docs/products/messaging/topics/+page.markdoc:407]: To use a model like `Subscriber` an additional import is needed 😬 

```dart
import 'package:dart_appwrite/models.dart';
```

File: src/routes/docs/products/messaging/send-sms-messages/+page.markdoc [hunk 3]
```diff
@@ -54,227 +54,227 @@ You can also implement forms in your app to collect contact information and add
...
+  identifier: '<IDENTIFIER>',
+  providerId: '<PROVIDER_ID>',  // (optional)
+  name: '<NAME>',               // (optional)
+);
+
+target.then((response) {
+  print(response);
+}).catchError((error) {
+  print(error.response);
+});
```
Comment (stnguyen90) [inline src/routes/docs/products/messaging/send-sms-messages/+page.markdoc:204]:
```suggestion
Target target = await users.createTarget(
  userId: '<USER_ID>',
  targetId: '<TARGET_ID>',
  providerType:  MessagingProviderType.phone,
  identifier: '<IDENTIFIER>',
  providerId: '<PROVIDER_ID>',  // (optional)
  name: '<NAME>',               // (optional)
);
```

File: src/routes/docs/products/messaging/send-sms-messages/+page.markdoc [hunk 4]
```diff
@@ -395,95 +369,74 @@ client = Client.new
...
 
-  client
-    .setEndpoint('https://cloud.appwrite.io/v1') // Your API Endpoint
-    .setProject('5df5acd0d48c2')                 // Your project ID
-    .setKey('919c2d18fb5d4...a2ae413da83346ad2') // Your secret API key
-  ;
+Messaging messaging = Messaging(client);
 
-  Future result = messaging.createTopic(
+Topic topic = messaging.createTopic(
```
Comment (stnguyen90) [inline src/routes/docs/products/messaging/send-sms-messages/+page.markdoc:403]:
```suggestion
Topic topic = await messaging.createTopic(
```

File: src/routes/docs/products/messaging/send-email-messages/+page.markdoc [hunk 5]
```diff
@@ -40,208 +40,432 @@ During development, you can add targets to existing accounts by navigating to **
...
+  identifier: '<IDENTIFIER>',
+  providerId: '<PROVIDER_ID>',  // (optional)
+  name: '<NAME>',               // (optional)
+);
+
+target.then((response) {
+  print(response);
+}).catchError((error) {
+  print(error.response);
+});
```
Comment (stnguyen90) [inline src/routes/docs/products/messaging/send-email-messages/+page.markdoc:195]:
```suggestion
final target = await users.createTarget(
  userId: '<USER_ID>',
  targetId: '<TARGET_ID>',
  providerType:  MessagingProviderType.email,
  identifier: '<IDENTIFIER>',
  providerId: '<PROVIDER_ID>',  // (optional)
  name: '<NAME>',               // (optional)
);
```

File: src/routes/docs/products/messaging/send-email-messages/+page.markdoc [hunk 6]
```diff
@@ -40,208 +40,432 @@ During development, you can add targets to existing accounts by navigating to **
...
+  identifier: '<IDENTIFIER>',
+  providerId: '<PROVIDER_ID>',  // (optional)
+  name: '<NAME>',               // (optional)
+);
+
+result.then((response) {
+  print(response);
+}).catchError((error) {
+  print(error.response);
+});
```
Comment (stnguyen90) [inline src/routes/docs/products/messaging/send-email-messages/+page.markdoc:423]:
```suggestion
import 'package:dart_appwrite/dart_appwrite.dart';
import 'package:dart_appwrite/enums.dart';
import 'package:dart_appwrite/models.dart';

Client client = Client()
  .setEndpoint('https://cloud.appwrite.io/v1')  // Your API Endpoint
  .setProject('5df5acd0d48c2')                  // Your project ID
  .setKey('919c2d18fb5d4...a2ae413da83346ad2'); // Your secret API key

Users users = Users(client);

Target result = await users.createTarget(
  userId: '<USER_ID>',
  targetId: '<TARGET_ID>',
  providerType:  MessagingProviderType.PHONE,
  identifier: '<IDENTIFIER>',
  providerId: '<PROVIDER_ID>',  // (optional)
  name: '<NAME>',               // (optional)
);
```

File: src/routes/docs/products/messaging/messages/+page.markdoc [hunk 7]
```diff
@@ -35,8 +35,8 @@ Each message displays with the following information.
 * Type
 * Type of message, either `Push`, `Email`, and `SMS`.
 ---
-* Status
-* Indicates the status of the message, can be one of `draft`, `scheduled`, `processing`, `failed`, `success`.
+* Draft
+* If the message is a draft or ready to be sent. Can be `true` or `false`.
```
Comment (stnguyen90) [inline src/routes/docs/products/messaging/messages/+page.markdoc:39]: Draft isn't shown in the table. The status is still shown

File: src/routes/docs/products/messaging/messages/+page.markdoc [hunk 8]
```diff
@@ -68,8 +68,8 @@ This helps you reach your user, even when their device do not have internet acce
 {% /table %}
 
 # Messages lifecycle {% #messages-lifecycle %}
-All messages begin as `draft`.
-If the message is sent immediately, its status is set to `processing`.
+All messages begin as `draft`.
```
Comment (stnguyen90) [inline src/routes/docs/products/messaging/messages/+page.markdoc:71]: Ooops...technically, messages can go straight to processing. Sorry!

**Review by stnguyen90 — CHANGES_REQUESTED** [review]
I made a couple comments about using `await` in node code samples, but there might be others outside of my comments.

I made a couple comments about dart and imports, but there may be other instances of the problem too. Maybe it would me sense to always have all 3 imports for simplicity:

```dart
import 'package:dart_appwrite/dart_appwrite.dart';
import 'package:dart_appwrite/enums.dart';
import 'package:dart_appwrite/models.dart';
```

How do we make sure the API references are updated?

File: src/routes/docs/products/messaging/messages/+page.markdoc (hunk 7 above)
Comment (gewenyu99) [reply src/routes/docs/products/messaging/messages/+page.markdoc:39]: Something's wrong with render againnnnn

File: src/routes/docs/products/messaging/messages/+page.markdoc (hunk 8 above)
Comment (gewenyu99) [reply src/routes/docs/products/messaging/messages/+page.markdoc]: Done

File: src/routes/docs/products/messaging/topics/+page.markdoc (hunk 1 above)
Comment (gewenyu99) [reply src/routes/docs/products/messaging/topics/+page.markdoc:302]: fixed

File: src/routes/docs/products/messaging/messages/+page.markdoc (hunk 7 above)
Comment (stnguyen90) [reply src/routes/docs/products/messaging/messages/+page.markdoc:39]: Sorry, I meant in the Console, there is no Draft column:

image

So the previous text was fine.

**Review by stnguyen90 — CHANGES_REQUESTED** [review]

**Review by ArmanNik — APPROVED** [review]

**Pull request merged**